datafusion/functions-nested/benches/array_has.rs

```diff
@@ -0,0 +1,377 @@
+            let needle_array = make_array(needle);
+
+            b.iter(|| black_box(array_has_all(list_array.clone(), needle_array.clone())))
+        },
+    );
+
+    // Benchmark: not all found (early exit)
+    group.bench_with_input(
+        BenchmarkId::new("early_exit", array_size),
+        &array_size,
+        |b, &size| {
+            let array = (0..size).map(|i| lit(i as i64)).collect::<Vec<_>>();
+            let list_array = make_array(array);
+            let needle_array = make_array(vec![lit(0_i64), lit(-1_i64)]); // -1 not in array
+
+            b.iter(|| black_box(array_has_all(list_array.clone(), needle_array.clone())))
+        },
+    );
+
+    group.finish();
+}
+
+fn bench_array_has_any(c: &mut Criterion, array_size: usize) {
+    let mut group = c.benchmark_group("array_has_any");
+
+    // Benchmark: first element matches (best case)
+    group.bench_with_input(
+        BenchmarkId::new("first_match", array_size),
+        &array_size,
+        |b, &size| {
+            let array = (0..size).map(|i| lit(i as i64)).collect::<Vec<_>>();
+            let list_array = make_array(array);
+            let needle_array = make_array(vec![lit(0_i64), lit(-1_i64), lit(-2_i64)]);
+
+            b.iter(|| black_box(array_has_any(list_array.clone(), needle_array.clone())))
+        },
+    );
+
+    // Benchmark: last element matches (worst case)
+    group.bench_with_input(
+        BenchmarkId::new("last_match", array_size),
+        &array_size,
+        |b, &size| {
+            let array = (0..size).map(|i| lit(i as i64)).collect::<Vec<_>>();
+            let list_array = make_array(array);
+            let needle_array = make_array(vec![lit(-1_i64), lit(-2_i64), lit(0_i64)]);
+
+            b.iter(|| black_box(array_has_any(list_array.clone(), needle_array.clone())))
+        },
+    );
+
+    // Benchmark: no match
+    group.bench_with_input(
+        BenchmarkId::new("no_match", array_size),
+        &array_size,
+        |b, &size| {
+            let array = (0..size).map(|i| lit(i as i64)).collect::<Vec<_>>();
+            let list_array = make_array(array);
+            let needle_array = make_array(vec![lit(-1_i64), lit(-2_i64), lit(-3_i64)]);
+
+            b.iter(|| black_box(array_has_any(list_array.clone(), needle_array.clone())))
+        },
+    );
+
+    group.finish();
+}
+
+fn bench_array_has_strings(c: &mut Criterion) {
+    let mut group = c.benchmark_group("array_has_strings");
+
+    // Benchmark with string arrays (common use case for tickers, tags, etc.)
+    let sizes = vec![10, 100, 1000];
+
+    for &size in &sizes {
+        group.bench_with_input(BenchmarkId::new("found", size), &size, |b, &size| {
+            let array = (0..size)
+                .map(|i| lit(format!("TICKER{i:04}")))
+                .collect::<Vec<_>>();
+            let list_array = make_array(array);
+            let needle = lit("TICKER0050");
```

Comment:
For the case of `size=10` this needle won't be in the array, so it will actually test `not_found`.

```suggestion
            let needle = lit("TICKER0005");
```